app/src/main/java/com/jerboa/ui/components/ban/BanFromCommunityActivity.kt

```diff
@@ -0,0 +1,132 @@
+package com.jerboa.ui.components.ban
+
+import android.util.Log
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Send
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.jerboa.JerboaAppState
+import com.jerboa.R
+import com.jerboa.api.ApiState
+import com.jerboa.communityNameShown
+import com.jerboa.datatypes.BanFromCommunityData
+import com.jerboa.db.entity.isAnon
+import com.jerboa.model.AccountViewModel
+import com.jerboa.model.BanFromCommunityViewModel
+import com.jerboa.personNameShown
+import com.jerboa.ui.components.common.ActionTopBar
+import com.jerboa.ui.components.common.getCurrentAccount
+
+object BanFromCommunityReturn {
+    const val BAN_DATA_VIEW = "ban-from-community::return(ban-data-view)"
+    const val BAN_DATA_SEND = "ban-from-community::send(ban-data-view)"
+}
+
+@Composable
+fun BanFromCommunityActivity(
```

Comment:
Very similar logic, so unfortunately a lot of copy pasting. But its better to have these activities separate and explicit IMO.